### ISSUE TITLE
Clean up ThreadModule

### ISSUE THREAD

**Issue by malinajirka** [wordpress-mobile/WordPress-Android#8975]
Replace `ThreadModule.MAIN_DISPATCHER` with `ThreadModule.UI_THREAD`. 

Consider replacing `IO_DISPATCHER` with `BG_THREAD` (which uses Dispatchers.Default) -> I somehow feel like explicitly stating whether we want a regular bg thread or thread for an IO operation is more clean.

**Comment by oguzkocer** [comment]
> We can definitely replace MAIN_DISPATCHER with UI_THREAD. We could theoretically also replace IO_DISPATCHER with BG_THREAD (which uses Dispatchers.Default), but I somehow feel like explicitly stating whether we want a regular bg thread or thread for an IO operation is more clean. Wdyt?

I thought it'd be better to reply to your comment here @malinajirka since that PR is now closed. I don't really mind either approach, but I think consistency across view models is important. So, maybe we should update our view models and use cases for now and once site creation is merged into `develop` we open a PR that changes all view models and use cases together if you think that'd be an improvement. That way we can also involve others in the decision. Would that work for you?

**Comment by oguzkocer** [comment]
Fixed in #9074.
